eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentInitiator.scala

```diff
@@ -81,16 +82,28 @@ class PaymentInitiator(nodeParams: NodeParams, router: ActorRef, register: Actor
 
     case pf: PaymentFailed => pending.get(pf.id).foreach(pp => {
       val decryptedFailures = pf.failures.collect { case RemoteFailure(_, Sphinx.DecryptedFailurePacket(_, f)) => f }
-      val canRetry = decryptedFailures.contains(TrampolineFeeInsufficient) || decryptedFailures.contains(TrampolineExpiryTooSoon)
-      pp.remainingAttempts match {
-        case (trampolineFees, trampolineExpiryDelta) :: remainingAttempts if canRetry =>
-          log.info(s"retrying trampoline payment with trampoline fees=$trampolineFees and expiry delta=$trampolineExpiryDelta")
-          sendTrampolinePayment(pf.id, pp.r, trampolineFees, trampolineExpiryDelta)
-          context become main(pending + (pf.id -> pp.copy(remainingAttempts = remainingAttempts)))
-        case _ =>
-          pp.sender ! pf
-          context.system.eventStream.publish(pf)
-          context become main(pending - pf.id)
+      val shouldRetry = decryptedFailures.contains(TrampolineFeeInsufficient) || decryptedFailures.contains(TrampolineExpiryTooSoon)
+      if (shouldRetry) {
+        pp.remainingAttempts match {
+          case (trampolineFees, trampolineExpiryDelta) :: remainingAttempts =>
+            log.info(s"retrying trampoline payment with trampoline fees=$trampolineFees and expiry delta=$trampolineExpiryDelta")
+            sendTrampolinePayment(pf.id, pp.r, trampolineFees, trampolineExpiryDelta)
+            context become main(pending + (pf.id -> pp.copy(remainingAttempts = remainingAttempts)))
```

Comment:
nit: `remainingAttempts = remainingAttempts` confused me a bit for a while. `remainingAttempts` could be renamed to `remainingTail` or something like that.

Reply:
Right, the same naming is a bit confusing. Updated in https://github.com/ACINQ/eclair/pull/1455/commits/3effac53b8ef5b33aefa4aa02d077035add75285